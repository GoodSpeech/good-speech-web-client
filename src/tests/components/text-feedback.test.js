import TextFeedbak from '../../components/text-feedback';
import { render } from '../test-utils';

const defaultProps = {
  textReadedFeedback: [{
    value: 'The',
    similarity: 0.7
  }, {
    value: ' cat is'
  }],
  textToRead: 'The cat is under the table.',
  textReaded: 'The cat is',
  interimText: 'under the',
  lang: 'en-US',
  onEditTextToRead: () => {},
  onTextToReadChange: () => {}
};

describe('TextFeedbak component', () => {
  it('Should display unsupported message', () => {
    const component = render(TextFeedbak, defaultProps);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Should call onEditTextToRead when focus content editable element', () => {
    const onEditTextToRead = jest.fn();
    const component = render(TextFeedbak, defaultProps, {onEditTextToRead});
    component.toJSON().children[0].props.onFocus();
    expect(onEditTextToRead).toBeCalled();
  });

  it('Should call onTextToReadChange when blur content editable element', () => {
    const onTextToReadChange = jest.fn();
    const component = render(TextFeedbak, defaultProps, {onTextToReadChange});
    component.toJSON().children[0].props.onBlur({
      currentTarget: {
        innerText: 'foo'
      }
    });
    expect(onTextToReadChange).toBeCalledWith('foo');
  });

  it('Should call onTextToReadChange when paste text in content editable element', () => {
    const onTextToReadChange = jest.fn();
    const component = render(TextFeedbak, defaultProps, {onTextToReadChange});
    component.toJSON().children[0].props.onPaste({
      preventDefault: () => {},
      clipboardData: {
        getData: () => 'bar'
      }
    });
    expect(onTextToReadChange).toBeCalledWith('bar');
  });

  it('Should add hover styles when the mouse enter to the content editor', () => {
    const component = render(TextFeedbak, defaultProps);
    component.toJSON().children[0].props.onMouseEnter();
    expect(component.toJSON()).toMatchSnapshot();
    component.toJSON().children[0].props.onMouseLeave();
    expect(component.toJSON()).toMatchSnapshot();
  });

});