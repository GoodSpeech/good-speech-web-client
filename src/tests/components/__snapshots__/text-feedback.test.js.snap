// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFeedbak component Should add hover styles when the mouse enter to the content editor 1`] = `
<div
  className="TextFeedback-container-2211185196"
>
  <p
    className="TextFeedback-text-3056024988"
    contentEditable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPaste={[Function]}
    suppressContentEditableWarning={true}
  >
    <span>
      <span
        className={undefined}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#999600",
          }
        }
      >
        The
      </span>
    </span>
    <span>
      <span
        className={undefined}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#008000",
          }
        }
      >
         cat is
      </span>
    </span>
    <span>
      <span
        className="TextFeedback-readed-604330111"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
         under the
      </span>
    </span>
    <span>
      <span
        className="TextFeedback-unreaded-1813682287"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
         table.
      </span>
    </span>
  </p>
  <div
    className="TextFeedback-editContainer-2585465578"
  >
    <span
      className="TextFeedback-editIconContainer-3256160467"
    >
      <i
        className="material-icons TextFeedback-editIcon-2721462721"
      >
        mode_edit
      </i>
       
      Edit
    </span>
  </div>
</div>
`;

exports[`TextFeedbak component Should add hover styles when the mouse enter to the content editor 2`] = `
<div
  className="TextFeedback-container-2211185196"
>
  <p
    className="TextFeedback-text-3056024988"
    contentEditable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPaste={[Function]}
    suppressContentEditableWarning={true}
  >
    <span>
      <span
        className={undefined}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#999600",
          }
        }
      >
        The
      </span>
    </span>
    <span>
      <span
        className={undefined}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#008000",
          }
        }
      >
         cat is
      </span>
    </span>
    <span>
      <span
        className="TextFeedback-readed-604330111"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
         under the
      </span>
    </span>
    <span>
      <span
        className="TextFeedback-unreaded-1813682287"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
         table.
      </span>
    </span>
  </p>
  <div
    className="TextFeedback-editContainer-2585465578"
  />
</div>
`;

exports[`TextFeedbak component Should display unsupported message 1`] = `
<div
  className="TextFeedback-container-2211185196"
>
  <p
    className="TextFeedback-text-3056024988"
    contentEditable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onPaste={[Function]}
    suppressContentEditableWarning={true}
  >
    <span>
      <span
        className={undefined}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#999600",
          }
        }
      >
        The
      </span>
    </span>
    <span>
      <span
        className={undefined}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "#008000",
          }
        }
      >
         cat is
      </span>
    </span>
    <span>
      <span
        className="TextFeedback-readed-604330111"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
         under the
      </span>
    </span>
    <span>
      <span
        className="TextFeedback-unreaded-1813682287"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={Object {}}
      >
         table.
      </span>
    </span>
  </p>
  <div
    className="TextFeedback-editContainer-2585465578"
  />
</div>
`;
